local-volume/provisioner/pkg/discovery/discovery.go

```diff
@@ -179,25 +179,24 @@ func (d *Discoverer) createPV(file, class string, config common.MountConfig, cap
 	glog.Infof("Created PV %q for volume at %q", pvName, outsidePath)
 }
 
-// roundDown the capacity so that it is easier to read
+// roundDown the capacity to an easy to read value
 func roundDownCapacityPretty(capacity int64) int64 {
-	// capacity is exactly divisible by Gi
-	if capacity%1024*1024*1024 == 0 {
-		return capacity
-	}
-	sizeGi := util.RoundUpSize(capacity, 1024*1024*1024)
-	if (sizeGi - 1) >= 10 {
-		// rounddown to nearest Gi
-		return (sizeGi - 1) * 1024 * 1024 * 1024
-	}
-	// capacity is exactly divisible by Mi
-	if capacity%1024*1024 == 0 {
-		return capacity
-	}
-	sizeMi := util.RoundUpSize(capacity, 1024*1024)
-	if (sizeMi - 1) >= 10 {
-		// rounddown to nearest Mi
-		return (sizeMi - 1) * 1024 * 1024
+
+	const (
+		KiB int64 = 1024
+		MiB int64 = 1024 * KiB
+		GiB int64 = 1024 * MiB
+		TiB int64 = 1024 * GiB
+	)
+	easyToReadSizes := []int64{TiB, GiB, MiB}
+
+	// rounddown to the nearest easy to read size
+	// such that there are at least 10 units at that size
+	for _, easyToReadSize := range easyToReadSizes {
+		size := util.RoundDownSize(capacity, easyToReadSize)
+		if size >= 10 {
```

Comment:
Can you add a test case that shows you had to go down to the next unit in order to keep the precision?

Reply:
done